### DISCUSSION TITLE
useCheckbox docs?

### DISCUSSION THREAD

**Discussion by ogrotten** [chakra-ui/chakra-ui#2933] [Q&A]
`useCheckbox` is mentioned on the Checkbox page, but it's not listed in the hooks section.

Is there a docs section for this hook?

**Comment by thomasbroderick** [comment]
Even if no docs are created, could we at least get a sample of how a Checkbox component might consume this hook?

**Reply by malashevskyi** [reply]
As I understood it is an internal hook, you probably don't need it at all, it is working under the hood.

![screenshot](https://i.imgur.com/02MYco7.png)

**Reply by tlylt** [reply]
It is working under the hood for display... but we need a way to retrieve the states when the checkbox is updated and respond to such events.

**Reply by malashevskyi** [reply]
> It is working under the hood for display... but we need a way to retrieve the states when the checkbox is updated and respond to such events.

Why you don't want to use `onChange`?

**Reply by tlylt** [reply]
I am ok with `onChange` + `useState`, just thought that if the `useCheckbox` does the exact same thing, I don't have to create my own state hook.

**Comment by ankursehdev** [comment]
One example here: https://github.com/chakra-ui/chakra-ui/issues/5943#issuecomment-1261245099
